### PR TITLE
Bug/smhe 1034 ophalen pq data gaat niet goed op kaifa meter

### PR DESCRIPTION
**Pull request by natasja-n**

### CODE REVIEW

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java [hunk 1]
```diff
@@ -188,4 +224,78 @@ private List<GetResult> createCaptureObjectsProfile2() {
 
     return Collections.singletonList(getResult);
   }
+
+  private List<GetResult> createPublicCaptureObjects() {
+    DataObjectDefinitions.getClockDefinition();
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java]: The result of this function is not used. Shouldn't the clock object be added to the getResult?

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java [hunk 2]
```diff
@@ -62,41 +76,63 @@ public void testHandlePrivateProfileSelectiveAccess() throws ProtocolAdapterExce
...
+            "PUBLIC",
+            Date.from(Instant.now().minus(2, ChronoUnit.DAYS)),
+            new Date(),
+            new ArrayList<>());
+
+    when(this.dlmsHelper.getAndCheck(
+            any(DlmsConnectionManager.class),
+            any(DlmsDevice.class),
+            any(String.class),
+            any(AttributeAddress.class)))
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java:115]: `Any` allows any list of adresses. If there is a duplicate clock entry, then this is not caught.

**Review by stefanermens — CHANGES_REQUESTED** [review]

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java (hunk 1 above)
Comment (natasja-n) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java]: Removed the unused line. The first structureData was the clock definition. I have updated the naming to clarify it is the clock definition.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java (hunk 2 above)
Comment (natasja-n) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileSelectiveAccessHandlerTest.java:115]: We have discussed this and decided to take this into account when refactoring the dataobjects.

**Review by stefanermens — APPROVED** [review]

**Pull request merged**